cli/add_metrics.php

```diff
@@ -0,0 +1,77 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+use tool_cloudmetrics\metric\test_metric;
+use tool_cloudmetrics\collector;
+
+/**
+ * Generate mock metric data to send to collectors.
+ *
+ * @package   tool_cloudmetrics
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2022, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+define('CLI_SCRIPT', true);
+
+require_once(__DIR__ . '/../../../../config.php');
+require_once($CFG->libdir.'/clilib.php');
+
+// Get cli options.
+list($options, $unrecognized) = cli_get_params(
+    [
+        'help' => false,
+        'name' => false,
+    ], [
+        'h' => 'help',
+        'n' => 'name',
+    ]
+);
+
+if ($unrecognized) {
+    $unrecognized = implode("\n  ", $unrecognized);
+    cli_error(get_string('cliunknowoption', 'admin', $unrecognized));
+}
+
+if ($options['help']) {
+    $help =
+        "Add metrics.
+
+Generate mock metric data to fill collectors with.
+Do not use with production environments.
```

Comment:
is it worth adding some checks here to prevent prod runs? We did this with datawash 5 different ways (really) to be safe, but it can be just 1 perhaps. Might not be worth.

There *may* be a security influence here, in that someone may be able to inject naughty data with a system compromise, and that may be used to divert monitoring attention to the fake load spike or similar, while the attacker targets another system.

Reply:
I added a simple $CFG check. You need to set a config value $CFG->config_php_settings['tool_cloudmetrics_allow_add_metrics'], to be able to run the script.
I basically copied check #4 from datacleaner.